### ISSUE TITLE
Current trunk version fails to compile on Windows

### ISSUE THREAD

**Issue by wolftobias** [openssl/openssl#1153]
Hi all,

today I tried to compile the latest trunk version and I got the following issues:

Microsoft (R) Program Maintenance Utility Version 11.00.61030.0
Copyright (C) Microsoft Corporation.  All rights reserved.

        cl -DOPENSSL_USE_APPLINK -DDSO_WIN32 -DNDEBUG -DOPENSSL_THREADS -DOPENSSL_NO_STATIC_ENGINE -DOPENSSL_PIC -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m
-DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DPOLY1305_ASM -DOPENSSL_API_COMPAT=0x10100000L "-DENGINESDIR=\"C:\\download\\openssl-1.
1.0-pre6\\openssl-master\\bin\\lib\\engines\"" "-DOPENSSLDIR=\"C:\\Program Files\\Common Files\\SSL\"" -W3 -wd4090 -Gs0 -GF -Gy -nologo -DOPENSSL_SYS_WIN32 -DWIN32_LEAN_AND_MEAN -DL_ENDIAN -D_CRT_SECU
RE_NO_DEPRECATE -DUNICODE -D_UNICODE /MD /O2  -rdynamic /Zi /Fdossl_static /I "include" /I "." /I "crypto\include" -c /Focrypto\mem_dbg.obj "crypto\mem_dbg.c"
cl : Command line warning D9002 : ignoring unknown option '-rdynamic'
mem_dbg.c
crypto\mem_dbg.c(20) : fatal error C1083: Cannot open include file: 'execinfo.h': No such file or directory
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 11.0\VC\BIN\x86_amd64\cl.EXE"' : return code '0x2'
Stop.

**Comment by mattcaswell** [comment]
Looks like a problem with the "enable-crypto-mdebug-backtrace" Configure option. I assume you selected that? Looks to be introduced by commit ef8ca6bd54. Ping @levitte.

**Comment by levitte** [comment]
Ah, yes, `crypto-mdebug-backtrace` is one of those "don't use unless you know what you're doing" 
It isn't implemented for native Windows, so don't use it there.

**Comment by mattcaswell** [comment]
Hmmmm....hang on. If its a don't use it on Windows thing, shouldn't we disable it there? Or at the very least document that it is unusable.

**Comment by wolftobias** [comment]
But I want to use it! I need the memory leaks detection mechanism very urgently!

**Comment by levitte** [comment]
@mattcaswell, yeah, you're right... In `INSTALL`, perhaps? Basically only available on systems that have execinfo (uhmmm, gnu based systems, iirc) 

@wolftobias, you can still use `crypto-mdebug`

**Comment by mattcaswell** [comment]
Yes in INSTALL is the right place - that's where "enable-crypto-mdebug-backtrace" is already documented. Is it possible for Configure to complain if you choose an option that is incompatible with your current platform?

**Comment by wolftobias** [comment]
But crypto-mdebug gives me just the info where the malloc was called, I have an issue with memory leaks where I want to use the new stack trace utility to get an idea where the caller comes from. Just to get more infos. I understood that this should be possible in the new version, right?

**Comment by mattcaswell** [comment]
Only on systems that support it :-(

**Comment by wolftobias** [comment]
which systems are supporting that? As far as I can see compiling on linux for win64 was working.

**Comment by levitte** [comment]
Like I said, GNU base system, or to be precise, it requires gcc *and* that it's implemented on the target system.

[the page on execinfo.h](https://www.gnu.org/software/gnulib/manual/html_node/execinfo_002eh.html) says that it's not implemented on mingw...  your experience tells me that has changed, or they have just implemented a stub on that system (which makes it a no-op).

**Comment by mattcaswell** [comment]
No that page says that gnulib fixes the problem where the header file is missing on some platforms (including mingw). It goes on to say that the implementation on such platforms is a stub and does nothing.

**Comment by levitte** [comment]
#1154

**Comment by levitte** [comment]
Uhmmm, @mattcaswell, I think this part of that page is quite clear about the availability of `execinfo.h`:

> <p>Portability problems fixed by Gnulib:</p>
> 
> <ul>
> <li>This header file is missing on some platforms: Mac OS X 10.3, FreeBSD 6.0, NetBSD 5.0, OpenBSD 3.8, Minix 3.1.8, AIX 5.1, HP-UX 11, IRIX 6.5, OSF/1 5.1, Solaris 10, Cygwin, mingw, MSVC 9, Interix 3.5, BeOS.</li>
> </ul>

That's part of the issue here, isn't it?  I didn't word that quite right in my previous comment no this, though.

**Comment by mattcaswell** [comment]
The key phrase is "Portability problems fixed by Gnulib", i.e. the portability problem *fixed* is that the header file is missing on mingw. A somewhat convoluted way of saying that *with* gnulib the header file is present on mingw.

The next section then says "Portability problems not fixed by Gnulib: On platforms where the header file is missing, the Gnulib substitute implementation is just a stub, and does nothing".

So in other words on mingw gnulib fixes the problem of a missing execinfo.h header file, but it fixes it by providing a stub that does nothing. So it will compile but it won't work.

**Comment by levitte** [comment]
Thanks

**Comment by mattcaswell** [comment]
So I think this really can be closed now in favour of #1154.
